packages/neon_framework/packages/account_repository/test/account_repository_test.dart

```diff
@@ -0,0 +1,564 @@
+      expect(
+        AccountRepository.isLogInQRCode(url: 'nc://login/user:JohnDoe&password:super_secret&server:example.com'),
+        isTrue,
+      );
+
+      expect(
+        AccountRepository.isLogInQRCode(url: 'invalid'),
+        isFalse,
+      );
+    });
+
+    group('loadAccounts', () {
+      test('no active account when no credentials are available', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => BuiltList());
+        await repository.loadAccounts();
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts, 'accounts', isEmpty)
+                .having((e) => e.active, 'active', isNull),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('emits stored credentials in accounts', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+        await repository.loadAccounts();
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts.map((e) => e.credentials), 'accounts', containsAllInOrder(accountList))
+                .having((e) => e.active, 'active', equals('b0682d652840ef50a4115cc77109bedd8c577ccc')),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('uses initialAccount as active', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+        await repository.loadAccounts(initialAccount: 'c73bc81eda746a735de697a0f0cbdb08f96ea3f3');
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts.map((e) => e.credentials), 'accounts', containsAllInOrder(accountList))
+                .having((e) => e.active, 'active', equals('c73bc81eda746a735de697a0f0cbdb08f96ea3f3')),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('ignores initialAccount when not it accounts', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+        await repository.loadAccounts(initialAccount: 'invalid');
+
+        expect(
+          repository.accounts,
+          emits(
+            isA<_AccountStream>()
+                .having((e) => e.accounts.map((e) => e.credentials), 'accounts', containsAllInOrder(accountList))
+                .having((e) => e.active, 'active', equals('b0682d652840ef50a4115cc77109bedd8c577ccc')),
+          ),
+        );
+
+        verifyNever(() => storage.readLastAccount());
+      });
+
+      test('uses last active account when specified', () async {
+        when(() => storage.readAccounts()).thenAnswer((_) async => accountList);
+        when(() => storage.readLastAccount()).thenAnswer((_) async => 'c73bc81eda746a735de697a0f0cbdb08f96ea3f3');
```

Comment:
Also verify the method was actually called once. It could still not get called and the first account was picked instead.